### ISSUE TITLE
Cursor timeout mechanism

### ISSUE THREAD

**Issue by afonit** [jupyterlab/jupyterlab#4384]
Is there a cursor timeout mechanism to stop the blinking that can be enabled somewhere?

I searched through the github issues and did not see a mention of one, at least with the words I was using to search.

Sometimes it is nice to just stay in position and sit and think about things as one is mulling things over.

An example of this is on the recent releases of fedora linux they have a terminal timeout where the cursor is still visible but stops blinking after 10 seconds.  The cycle then restarts after you move the cursor and stop.

Current workarounds:
* I can currently hit esc to go into command mode for the cell and that will take the cursor away.
* I use the vim plugin and do colon q to go to command mode.

I am sure it can be reasonable to expect a user to just use the two methods I mentioned above - though I thought just having a timeout might be a nice touch if it is not implemented.

**Comment by jasongrout** [comment]
That's a really cool idea. We use codemirror for our editor, so I would look in codemirror to see if it has the feature that we can expose.

It does look like codemirror exposes the blink rate: https://codemirror.net/doc/manual.html#config

Likely we'd need to implement the timeout ourselves.

**Comment by jasongrout** [comment]
Feel free to work on this if you want. It may involve changes to the codeeditor package to expose an interface for the blinking.

**Comment by jwkvam** [comment]
There's a configuration `cursorBlinkRate` documented [here](https://codemirror.net/doc/manual.html). I wouldn't mind if the blinking could be permanently disabled. That's how I have my vim configured.

**Comment by afonit** [comment]
While researching this I came across a post online that showed this:
```
%%javascript
CodeMirror.defaults.cursorBlinkRate=0
```
Which works in the classic notebook (non jupyterlab)

However in jupyterlab one gets the error:
```
JavaScript output is disabled in JupyterLab
```

I know altair/bokeh are getting around this by making an extension.

Would an extension make sense here?

**Comment by mikeivanov** [comment]
@jasongrout does "removed this from the Future" mean that it won't be implemented?

**Comment by ian-r-rose** [comment]
No, we're just doing some issue maintenance this week, reorganizing milestones, etc.

**Comment by gloryVine** [comment]
Any news on this? I'm using a vim-plugin and having a big green blinking cursor in normal mode mades JupyterLab unusable to me.
